src/pmdas/denki/denki.c

```diff
@@ -594,21 +695,24 @@ denki_label(int ident, int type, pmLabelSet **lpp, pmdaExt *pmda)
 	switch (type) {
 		case PM_LABEL_INDOM:
 			serial = pmInDom_serial((pmInDom)ident);
-			if (serial == RAPLRATE_INDOM) {
-				pmdaAddLabels(lpp, "{\"indom_name\":\"raplrate\"}");
-			}
-			if (serial == RAPLRAW_INDOM) {
-				pmdaAddLabels(lpp, "{\"indom_name\":\"raplraw\"}");
+			switch (serial) {
+				case RAPLRATE_INDOM:
+					pmdaAddLabels(lpp, "{\"indom_name\":\"raplrate\"}");
+					break;
+				case RAPLRAW_INDOM:
+					pmdaAddLabels(lpp, "{\"indom_name\":\"raplraw\"}");
+					break;
+				case ENERGYNOWRAW_INDOM:
+					pmdaAddLabels(lpp, "{\"indom_name\":\"energynowraw\"}");
+					break;
+				case ENERGYNOWRATE_INDOM:
+					pmdaAddLabels(lpp, "{\"indom_name\":\"energynowrate\"}");
+					break;
+				case POWERNOW_INDOM:
+					pmdaAddLabels(lpp, "{\"indom_name\":\"powernow\"}");
+					break;
 			}
 			break;
-		case PM_LABEL_ITEM:
-			if (pmID_cluster(ident) != 1)
-				break;
-			if (pmID_item(ident) == 0)	/* energy_now_raw */
-				pmdaAddLabels(lpp, "{\"units\":\"watt hours\"}");
-			if (pmID_item(ident) == 1)	/* energy_now_rate */
-				pmdaAddLabels(lpp, "{\"units\":\"watts\"}");
-			break;
```

Comment:
Hmm - this seems like useful units information that we're losing here - was it not correct?

Reply:
Right, was just taking over the rapl indoms here - but denki.bat has proper units.